service/src/main/java/aero/minova/cas/service/QueueService.java

```diff
@@ -227,88 +252,117 @@ private void saveMessage(String message, String procedureName, String topic) {
-		messagesRow.addValue(null);
-		messagesRow.addValue(null);
-		messagesRow.addValue(null);
-		messagesRow.addValue(null);
-		messagesRow.addValue(null);
-		messagesRow.addValue(null);
-		messagesRow.addValue(null);
-		messagesRow.addValue(null);
-
-		messagesRequest.addRow(messagesRow);
-		try {
-			unsendMessages = securityService.unsecurelyGetIndexView(messagesRequest);
+			serviceMessageRepo.saveAndFlush(pendingMessage);
+			logger.logQueueService("Deleting message with key " + pendingMessage.getKeylong());
 		} catch (Exception e) {
-			logger.logError("Could not read next message. The QueueService could not access the view " + messagesRequest.getName(), e);
+			logger.logError("The message with key " + pendingMessage.getKeylong() + " could not be deleted!", e);
 			throw new RuntimeException(e);
 		}
-		return unsendMessages;
 	}
 
 	/**
 	 * Versendet eine Nachricht.
 	 * 
-	 * @param nextMessage
+	 * @param pendingMessage
 	 *            Eine Row, in der folgende Übergabeparamtern enthalten sind:CASServiceKey, CASServiceName, ServiceURL, Port, MessageKey, Message, isSent,
 	 *            NumberOfAttempts, MessageCreationDate
 	 * @return true, falls der Versandt erfolgreich war. Andernfalls false.
 	 */
-	private boolean sendMessage(Row nextMessage) {
-		RestTemplate restTemplate = new RestTemplate();
-		String url = nextMessage.getValues().get(2).getStringValue() + ":" + nextMessage.getValues().get(3).getIntegerValue();
-		String message = nextMessage.getValues().get(5).getStringValue();
-
-		HttpEntity<?> request = new HttpEntity<Object>(message);
-		logger.logQueueService("Trying to send message with key " + nextMessage.getValues().get(4).getIntegerValue() + " to " + url);
+	private boolean sendMessage(ServiceMessage pendingMessage) {
+		// URL + : + Port
+		String url = pendingMessage.getCasservice().getPort() != 0
+				? pendingMessage.getCasservice().getServiceurl() + ":" + pendingMessage.getCasservice().getPort()
+				: pendingMessage.getCasservice().getServiceurl();
+		String message = pendingMessage.getMessage();
+
+		int serviceMessageReceiverLoginTypeKey = pendingMessage.getCasservice().getReceiverLoginType() != null
+				? pendingMessage.getCasservice().getReceiverLoginType().getKeylong()
+				: 0;
+
+		HttpEntity<?> request;
 		try {
+
+			// Falls BasicAuth:
+			if (serviceMessageReceiverLoginTypeKey == 2) {
+
+				// Username + : + Password
+				String credentials = pendingMessage.getCasservice().getUsername() + ":" + pendingMessage.getCasservice().getPassword();
+
+				HttpHeaders header = new HttpHeaders();
+				byte[] encodedAuth = Base64.encodeBase64(credentials.getBytes(StandardCharsets.UTF_8), false);
+				header.add("Authorization", "Basic " + encodedAuth);
+
+				request = new HttpEntity<>(message, header);
+
+				// Falls OAuth2:
+			} else if (serviceMessageReceiverLoginTypeKey == 3) {
+
+				String token;
+				CASServices service = pendingMessage.getCasservice();
+				if (!oauth2TokenList.containsKey(service)) {
```

Comment:
Müsste man hier nicht prüfen, ob der Token in den bspw. nächsten 10 Sekunden ungültig wird,
und in diesem Fall einen neuen Token anfordern?
Ansonsten könnte man `OAuth2Token#expiryDate` auch weglassen,
um den Code einfacher zu halten.

Reply:
Oh man, ja, du hast Recht.

Deswegen habe ich eigentlich den OAuth2Token eingeführt ^^'